dotnet/src/Connectors/Connectors.Amazon/Bedrock/Core/Clients/BedrockChatCompletionClient.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Runtime.CompilerServices;
+using Amazon.BedrockRuntime;
+using Amazon.BedrockRuntime.Model;
+using Connectors.Amazon.Bedrock.Core;
+using Connectors.Amazon.Models;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace Microsoft.SemanticKernel.Connectors.Amazon.Core;
+
+/// <summary>
+/// Represents a client for interacting with the chat completion through Bedrock.
+/// </summary>
+public class BedrockChatCompletionClient
+{
+    private readonly string _modelId;
+    private readonly string _modelProvider;
+    private readonly IAmazonBedrockRuntime _bedrockApi;
+    private readonly IBedrockModelIOService _ioService;
+    private Uri? _chatGenerationEndpoint;
+
+    /// <summary>
+    /// Builds the client object and registers the model input-output service given the user's passed in model ID.
+    /// </summary>
+    /// <param name="modelId"></param>
+    /// <param name="bedrockApi"></param>
+    /// <exception cref="ArgumentException"></exception>
+    protected BedrockChatCompletionClient(string modelId, IAmazonBedrockRuntime bedrockApi)
+    {
+        this._modelId = modelId;
+        this._bedrockApi = bedrockApi;
+        this._ioService = new BedrockClientIOService().GetIOService(modelId);
+        this._modelProvider = new BedrockClientIOService().GetModelProvider(modelId);
+    }
+    /// <summary>
+    /// Builds the convert request body given the model ID (as stored in ioService object) and calls the ConverseAsync Bedrock Runtime action to get the result.
+    /// </summary>
+    /// <param name="converseRequest"> The converse request for ConverseAsync bedrock runtime action. </param>
+    /// <param name="cancellationToken"> A cancellation token to cancel the operation. </param>
+    /// <returns></returns>
+    private async Task<ConverseResponse> ConverseBedrockModelAsync(
+        ConverseRequest converseRequest,
+        CancellationToken cancellationToken = default)
+    {
+        // Check that the text from the latest message in the request object is not empty.
+        string? text = converseRequest.Messages?[^1]?.Content?[0]?.Text;
+        if (string.IsNullOrWhiteSpace(text))
+        {
+            throw new ArgumentException("Did not enter proper chat completion message. Text was null or whitespace.");
```

Comment:
So this method checks that the latest message in the `converseRequest` isn't null or empty. Are we doing the same for Streaming?

Should we just do that inside `public static List<Message> BuildMessageList(ChatHistory chatHistory)` instead so it's leveraged from both places?

The logic flows between `internal async Task<IReadOnlyList<ChatMessageContent>> GenerateChatMessageAsync` and `internal async IAsyncEnumerable<StreamingChatMessageContent> StreamChatMessageAsync` seem inconsistent where it should be more consistent.